src/feedback/mod.rs

```diff
@@ -356,6 +395,57 @@ pub trait FciParser<'a>: Sized {
     fn parse(data: &'a [u8]) -> Result<Self, RtcpParseError>;
 }
 
+/// Stack or heap allocated [`FciBuilder`] wrapper.
+///
+/// This wrapper allows borrowing or owning an [`FciBuilder`] without
+/// propagating the concrete type of the [`FciBuilder`] to types such as
+/// [`PayloadFeedbackBuilder`] or [`TransportFeedback`]. This is needed for
+/// these types to be included in collections such as [`crate::CompoundBuilder`].
+///
+/// `Cow` from `std` would not be suitable here because it would require
+/// declaring the `B` type parameter as an implementer of `ToOwned`,
+/// which is not object-safe.
+#[derive(Debug)]
+enum FciBuilderWrapper<'a> {
+    Borrowed(&'a dyn FciBuilder<'a>),
+    Owned(Box<dyn FciBuilder<'a>>),
```

Comment:
```suggestion
    Owned(Box<dyn FciBuilder<'static>>),
```

? Otherwise seems to kind of defeat the point

Reply:
Changed the `builder_owned()` constructors to accept `fci: impl FciBuilder<'static> + 'static` and added a comment to explain why I used `Owned(Box<dyn FciBuilder<'a>>)`:

>     // Note: using `'a` and not `'static` here to help with the implementations
>     // of `deref()` and `as_ref()`. This enum is private and the `Owned` variant
>     // can only be constructed from `PayloadFeedbackBuilder::builder_owned` &
>     // `TransportFeedback::builder_owned` which constrain the `FciBuilder` to be `'static`.